insights/tools/query.py

```diff
@@ -130,28 +131,51 @@ def get_datasources():
 
 
 def matches(d, path):
-    def _simple_file(src):
-        if src.startswith(('insights_commands', 'sos_commands')):
-            src = src.split('/')[-1].split('_')[0]
-        return src
+    path = path.strip('/')
 
     if isinstance(d, sf.simple_file):
-        return path.strip("/") in _simple_file(d.path.strip('/').split()[0])
+        # Set the filename from the simple_file string as search target
+        search_tgt = d.path.strip('/').split()[0].split('/')[-1]
+        return search_tgt.startswith(path)
 
     if isinstance(d, sf.glob_file):
-        return any(re.match(path, glob2re(_simple_file(pat))) and not d.ignore_func(path) for pat in d.patterns)
+        for pat in d.patterns:
+            # Set the last part of the glob_file as the search target
+            pat_sp = pat.split('/')
+            search_tgt = pat_sp[-1]
+            if search_tgt.startswith("*") and len(pat_sp) >= 2:
+                # if the last part is "*" or "*XXX", set the parent dir as search target
+                # e.g: xinetd_conf: "/etc/xinet.d/*.conf"
+                #      logrotate_conf: "/etc/logrotate.d/*"
```

Comment:
Can we give relevant parser information when look up the qemu vm configs in `/etc/libvirt/qemu/`?

```
qemu_xml = glob_file(r"/etc/libvirt/qemu/*.xml")
```

 For example, want to analyze a file `rhel7u2.xml`. Or we should use the `qemu` as a key word for searching?

```
$ insights-info qemu
insights.parsers.qemu_xml.QemuXML
insights.parsers.qemu_conf.QemuConf

$ insights-info rhel7u2.xml
==>NULL
```

Reply:
@lonicerae - Good catching. 
But in fact, this situation has been considered. And, the `insights-info qemu` is the current solution for this situation. 
Think of the `*.conf`, currently, its candidate specs are as follows:
~~~
$ grep "\*.conf" insights/specs/default.py
    etc_journald_conf_d = glob_file(r"etc/systemd/journald.conf.d/*.conf")
    httpd_conf = glob_file(["/etc/httpd/conf/httpd.conf", "/etc/httpd/conf.d/*.conf"])
    httpd_conf_scl_httpd24 = glob_file(["/opt/rh/httpd24/root/etc/httpd/conf/httpd.conf", "/opt/rh/httpd24/root/etc/httpd/conf.d/*.conf"])
    httpd_conf_scl_jbcs_httpd24 = glob_file(["/opt/rh/jbcs-httpd24/root/etc/httpd/conf/httpd.conf", "/opt/rh/jbcs-httpd24/root/etc/httpd/conf.d/*.conf"])
    krb5 = glob_file([r"etc/krb5.conf", r"etc/krb5.conf.d/*.conf"])
    limits_conf = glob_file(["/etc/security/limits.conf", "/etc/security/limits.d/*.conf"])
    modprobe = glob_file(["/etc/modprobe.conf", "/etc/modprobe.d/*.conf"])
    pluginconf_d = glob_file("/etc/yum/pluginconf.d/*.conf")
    sysctl_conf_initramfs = simple_command("/bin/lsinitrd /boot/initramfs-*kdump.img -f /etc/sysctl.conf /etc/sysctl.d/*.conf")
    tmpfilesd = glob_file(["/etc/tmpfiles.d/*.conf", "/usr/lib/tmpfiles.d/*.conf", "/run/tmpfiles.d/*.conf"])
    usr_journald_conf_d = glob_file(r"usr/lib/systemd/journald.conf.d/*.conf")  # note that etc_journald.conf.d also exists
    virt_who_conf = glob_file([r"etc/virt-who.conf", r"etc/virt-who.d/*.conf"])
~~~
It's not a good experience to output all the `unrelated` parsers when querying `$ insights-info abc.conf`.

Reply:
@lonicerae @xiangce - there are a bunch of files with ".xml" suffix. 
~~~
./sos_commands/sar/sa16.xml
./etc/firewalld/zones/public.xml
./etc/libvirt/qemu/networks/autostart/default.xml
./etc/libvirt/qemu/networks/default.xml
~~~
Running "insights-info rhel7u2.xml" may trigger wrong parsers unless we consider to use full path. This only happens to glob-typed specs like:

~~~
etc_journald_conf_d = glob_file(r"etc/systemd/journald.conf.d/*.conf")
initscript = glob_file(r"etc/rc.d/init.d/*
krb5 = glob_file([r"etc/krb5.conf", r"etc/krb5.conf.d/*.conf"])
ovirt_engine_confd = glob_file("/etc/ovirt-engine/engine.conf.d/*")
yum_repos_d = glob_file("/etc/yum.repos.d/*")
...
~~~

TBH, I don't have any better method on my mind to better @xiangce's implementation. Let's have a quick discussion today.